src/librustc/middle/stability.rs

```diff
@@ -237,6 +237,113 @@ impl<'a, 'tcx> Annotator<'a, 'tcx> {
             }
         }
     }
+
+    fn annotate_no_inherit<F>(&mut self, hir_id: HirId, attrs: &[Attribute],
```

Comment:
Why "no inherit"? Seems harmless to use the usual method and reduce code duplication.

Reply:
Inheriting stability and deprecations caused rustc not to build. I suspect that it could have been made to work, by adding duplicate ignore directives on each type argument to deprecated names, but that didn't seem like a good experience.

Reply:
What exactly went wrong? I would also have thought `annotate` should work.